src/Server.js

```diff
@@ -52,6 +53,12 @@ class Server extends EventEmitter {
      */
     this.port = serverArgs.port || null;
 
+    /**
+     * The headers to use in requests
+     * @type {Object}
+     */
+    this.headers = serverArgs.headers || null;
```

Comment:
If we set this to an empty object instead of null could we avoid all the if checks and just call req.set({})? That might make the change cleaner for me.

Reply:
I was thinking about that as well - we should just double check that it works (it should!)